pkg/didcomm/protocol/introduce/service.go

```diff
@@ -201,6 +210,30 @@ func (s *Service) doHandle(msg *service.DIDCommMsg, outbound bool) (*metaData, e
 	}, nil
 }
 
+func (s *Service) handleInboundRequest(msg *service.DIDCommMsg, aEvent chan<- service.DIDCommAction) error {
+	thID, err := msg.ThreadID()
+	if err != nil {
+		return err
+	}
+
+	aMsg := s.newDIDCommActionMsg(&metaData{
+		record: record{
+			StateName: stateNameStart,
+			WaitCount: initialWaitCount,
+		},
+		Msg:      msg,
+		ThreadID: thID,
+	})
+	// there is no way to execute Continue or Stop functions
+	// RequestMsgType is not a part of the state machine
+	// and Client should handle this message type and act according to that (send a proposal)
+	aMsg.Continue = func(args interface{}) { panic("action is forbidden") }
+	aMsg.Stop = func(err error) { panic("action is forbidden") }
+	aEvent <- aMsg
```

Comment:
@llorllale @troyronda @rolsonquadras  What do you think about it?
When we received a Request message we cannot proceed with the protocol. First of all Request message is not a part of the state machine (from the doc). The request message is a kind of ping message or trigger message, but we should not respond to that. There is no even response message for Request. So, my suggestion is to forbid this execution (Continue()), and instead of calling `Continue()` client should send a proposal that does not have anything in common with the request message. 
e.g When Client receives a Request message he may:
* ignore that message
* send a proposal to any agent
* call Continue() and get panic (next time he will handle it properly :))

Reply:
doing panic means the whole agent will go down... I get that this is not supposed to happen.. but is there a way to tell the agent (or support team) what's the root cause of this state? (maybe simply logging the error with a meaningful message is enough)

But providing hints to the ops, it will be easier to investigate what went wrong. I'm not familiar with the state machine, but adding hints will definitely help. And personally, I would avoid panics unless the agent is genuinely in a non recoverable state. 

I guess @troyronda @llorllale @rolsonquadras might have a better opinion about it.

Reply:
First of all this comment related to the design (panic or logging is not a problem), the question is should we proceed with protocol after receiving this message (`Continue()` or `Stop()`) or forbid it (not possible to use `Continue()` or `Stop()`) and start sending a proposal independently (no matter what we received in the Request)

Reply:
@soluchok 

> When we received a Request message we cannot proceed with the protocol. [...] Request message is not a part of the state machine (from the doc)

I opened hyperledger/aries-rfcs#309 regarding this - I have my doubts.

Reply:
We should not have panics.

Reply:
@troyronda @Baha-sk Done!

Reply:
@soluchok I pushed PR hyperledger/aries-rfcs#313 to address hyperledger/aries-rfcs#309. Basically, there is agreement that the request message *is* part of the state machine.

Reply:
@llorllale just to clarify `requesting` state is still optional right?

Reply:
@soluchok yes, the introducee can begin from either the requesting state or the deciding state